File: hw2/src/train.cpp
#include "train.h"

Train::Train(int speed, float distance, int carsCount)
    : BaseTransport(speed, distance), m_carsCount(carsCount)
{
}

void Train::Out(std::ostream& out)
{
    BaseTransport::Out(out);
    out << ", cars count = " << m_carsCount << std::endl;
}

std::string_view Train::GetName()
{
    return "Train";
}

File: hw3/src/train.py
'''
train.py - содержит описание поезда
'''

from base_transport import *

# Поезд
class Train(BaseTransport):
    # Инициализация
    def __init__(self, speed, distance, cars_count):
        super().__init__(speed, distance)
        self.cars_count = cars_count

    # Конвертация в строку
    def __str__(self):
        return super().__str__() + f', cars_count = {self.cars_count}'
